python/lsst/faro/measurement/DetectorTableMeasurement.py

```diff
@@ -0,0 +1,98 @@
+                                          defaultTemplates={"refDataset": ""}):
+
+    catalog = pipeBase.connectionTypes.Input(
+        doc="Source table in parquet format, per visit",
+        dimensions=("instrument", "visit", "band"),
+        storageClass="DataFrame",
+        name="sourceTable_visit",
+        deferLoad=True
+    )
+
+    measurement = pipeBase.connectionTypes.Output(
+        doc="Per-detector measurement",
+        dimensions=("instrument", "visit", "detector", "band"),
+        storageClass="MetricValue",
+        name="metricvalue_{package}_{metric}"
+    )
+
+
+class DetectorTableMeasurementConfig(CatalogMeasurementBaseConfig,
+                                     pipelineConnections=DetectorTableMeasurementConnections):
+    """Configuration for DetectorTableMeasurementTask."""
+
+    columns = pexConfig.ListField(doc="Columns from sourceTable_visit to load.",
+                                  dtype=str, default=['coord_ra', 'coord_dec', 'detector'])
+
+    def validate(self):
+        super().validate()
+        if 'detector' not in self.columns:
+            msg = "The column `detector` must be appear in the list of columns."
+            raise pexConfig.FieldValidationError(DetectorTableMeasurementConfig.columns, self, msg)
+
+
+class DetectorTableMeasurementTask(CatalogMeasurementBaseTask):
+    """Base class for science performance metrics measured on single-detector source catalogs."""
+
+    ConfigClass = DetectorTableMeasurementConfig
+    _DefaultName = "detectorTableMeasurementTask"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        catalog = inputs['catalog'].get(parameters={'columns': self.config.columns})
+        selection = (catalog['detector'] == butlerQC.quantum.dataId['detector'])
+        catalog = catalog[selection]
+
+        kwargs = {}
+        if self.config.connections.refDataset != '':
+            refCats = inputs.pop('refCat')
+            filterList = [butlerQC.quantum.dataId.records['physical_filter'].name]
+            # Time at the start of the visit
+            epoch = butlerQC.quantum.dataId.records['visit'].timespan.begin
+            refCat, refCatCorrected = self._getReferenceCatalog(butlerQC,
+                                                                [ref.datasetRef.dataId
+                                                                 for ref in inputRefs.refCat],
+                                                                refCats,
+                                                                filterList,
+                                                                epoch)
+            kwargs['refCat'] = refCat
+            kwargs['refCatCorrected'] = refCatCorrected
+
+        outputs = self.run(catalog, **kwargs)
+        if outputs.measurement is not None:
+            butlerQC.put(outputs, outputRefs)
+        else:
+            self.log.debug("Skipping measurement of {!r} on {} "
```

Comment:
I'm leaning towards this being an `info` level message, since otherwise things get skipped and you never know about it in normal processing.

Reply:
This relates to another discussion we have been having about how to report various potential failure states. @jeffcarlin may be interested. I suggest we keep this as `debug` level for now, because it is the convention across the `faro` package at the moment, and if we want to make this change, we should make it in multiple places.